### ISSUE TITLE
video pass through possible also?

### ISSUE THREAD

**Issue by BugMagnet** [RandomEngy/VidCoder#18]
Over the past couple decades I have accumulated a sizable collection of videos, many/most in .AVI format. While I enjoy these on my computer with VLC player, I recently obtained a new "smart" TV with ROKU built in. I was dismayed to learn that the ROKU video player is very crippled as it doesn't recognize .AVI files at all. It seems to be very picky, requiring MKV, MP4 or MOV containers and h264 video with AAC, AC3 or MP3 audio (It doesn't play DTS even!)

I am a newbie at conversions and my research lead me to handbrake and vidcoder. I noticed that HB and vidcoder have a conditional audio passthrough, so I can select AAC, AC3 and MP3 to pass without transcoding. 

  * AAC, MP3, Vorbis, AC3, FLAC audio encoding and AAC/AC3/MP3/DTS/DTS-HD passthrough

With handbrake, I see an option to passthrough several different formats, with others being transcoded as needed. I am not all that familiar yet with vidcoder but it appears to limit audio passthrough to a single format, that chosen for re-encoding. Am I understand that right?

For my purposes, this ROKU player will accept AAC, AC3 and MP3 so my choice would be to pass all of these through and just convert other formats to either AAC or AC3. Can this be done with the current version/beta of vidcoder?

Now the biggie...Since video transcoding is the most time consuming I am thinking a similar pass-through for that would be great. I can't see it in handbrake and therefore I assume vidcoder can't provide that right?

Basically what I am after is a test, for example of an .AVI file. If it already contains h264 video and (AAC,AC3,MP3) audio, simply repackage in an MKV container. Am I on the right track with this? 

Since handbrake will never support video pass-through functionality, might it be possible that vidcoder could redirect such muxing to a different engine??

**Comment by RandomEngy** [comment]
Handbrake is rather integrated; I don't think it's possible to hook in and swap out the muxer for something else. If you need a pure muxer I think you're better off looking for another program.
